ginivision/src/main/java/net/gini/android/vision/camera/CameraFragmentImpl.java

```diff
@@ -977,70 +980,53 @@ public void documentRejected(@NonNull final String messageForUser) {
+                    mInMultiPageState = true;
+                    mMultiPageDocument = new ImageMultiPageDocument(true);
                 }
-            } else {
-                String errorMessage = context.getString(
-                        R.string.gv_document_import_invalid_document);
-                final FileImportValidator.Error error = fileImportValidator.getError();
-                if (error != null) {
-                    errorMessage = context.getString(error.getTextResource());
+                mMultiPageDocument.addDocuments(multiPageDocument.getDocuments());
+                if (mMultiPageDocument.getDocuments().isEmpty()) {
+                    LOG.error("Document import failed: Intent did not contain images");
+                    showInvalidFileError(null);
+                    mMultiPageDocument = null;
+                    mInMultiPageState = false;
+                    return;
                 }
-                addMultiPageDocumentError(errorMessage);
-            }
-        }
-        if (mMultiPageDocument.getDocuments().isEmpty()) {
-            LOG.error("Document import failed: Intent did not contain images");
-            showInvalidFileError(null);
-            mMultiPageDocument = null;
-            mInMultiPageState = false;
-            return;
-        }
-        LOG.info("Document imported: {}", mMultiPageDocument);
-        showActivityIndicatorAndDisableInteraction();
+                LOG.info("Document imported: {}", mMultiPageDocument);
 
-        updateImageStack();
+                updateImageStack();
 
-        final View view = mFragment.getView();
-        if (view == null) {
-            return;
-        }
-        view.postDelayed(new Runnable() {
-            @Override
-            public void run() {
-                requestClientDocumentCheck(mMultiPageDocument);
+                final View view = mFragment.getView();
+                if (view == null) {
+                    return;
+                }
+                view.postDelayed(new Runnable() {
+                    @Override
+                    public void run() {
+                        requestClientDocumentCheck(mMultiPageDocument);
+                    }
+                }, ImageStack.ADD_IMAGE_TRANSITION_DURATION_MS);
             }
-        }, ImageStack.ADD_IMAGE_TRANSITION_DURATION_MS);
-    }
 
-    private void addMultiPageDocumentError(final String string) {
-        final ImageDocument document = DocumentFactory.newEmptyImageDocument();
-        mMultiPageDocument.addDocument(document);
-        final GiniVisionDocumentError documentError = new GiniVisionDocumentError(string);
-        mMultiPageDocument.addErrorForDocument(document, documentError);
+            @Override
+            public void onError(final Exception exception) {
+                LOG.error("Document import failed", exception);
+                hideActivityIndicatorAndEnableInteraction();
+                showInvalidFileError(null);
```

Comment:
I would rather to see here a generic/general/whatever `FileImportValidator.Error` with the resource `R.string.gv_document_import_invalid_document` rather than `null`. But I'm ok if you decide to keep it.